### DISCUSSION TITLE
Keycloak integration tests fail on PPC64le

### DISCUSSION THREAD

**Discussion by Chandranana** [keycloak/keycloak#24856] [Q&A] [answered]
I  have tried building keycloak on ppc64le, was able to build it successfully. To fix other test failures we have installed power supported phantomjs binaries using below steps:
> https://github.com/ibmsoe/phantomjs/releases/download/2.1.1/phantomjs-2.1.1-linux-ppc64.tar.bz2
> tar -xvf phantomjs-2.1.1-linux-ppc64.tar.bz2 -C /usr/local 
> export PATH=$PATH:/usr/local/phantomjs-2.1.1-linux-ppc64/bin

However some of the integration tests have failed from org.keycloak.testsuite.x509.**. Below is the log for failing tests: 

```
-- [ERROR]   Run 1: X509OCSPResponderTest>AbstractTestRealmKeycloakTest.deleteCookies:84->AbstractKeycloakTest.deleteAllCookiesForRealm:291 » WebDriver The PhantomJS/GhostDriver server has unexpectedly died!
-- Build info: version: '3.14.0', revision: 'aacccce0', time: '2018-08-02T20:19:58.91Z'
-- System info: host: 'apache-kafka-1', ip: '127.0.1.1', os.name: 'Linux', os.arch: 'ppc64le', os.version: '5.4.0-155-generic', java.version: '17.0.8'
-- Driver info: driver.version: RemoteWebDriver
```

Please suggest if we can fix this. 

Also for some of the tests intermittently getting below error:
```
-- [ERROR] org.keycloak.testsuite.x509.X509OCSPResponderTest.loginFailedOnOCSPResponderRevocationCheckWithoutCA
-- [ERROR]   Run 1: X509OCSPResponderTest>AbstractKeycloakTest.beforeAbstractKeycloakTest:184->AbstractX509AuthenticationTest.importTestRealms:316->AbstractKeycloakTest.importTestRealms:392->AbstractKeycloakTest.importRealm:512 » Processing jakarta.ws.rs.NotAuthorizedException: HTTP 401 Unauthorized
```

Please suggest if this can be resolved too.

**Comment by jonkoops** [comment, marked as answer]
We've been meaning to get rid of PhantomJS in our test suite since quite some time, which is the most permanent and comprehensive solution for this issue. If you want to help out with that you can volunteer for #9979.

**Reply by Chandranana** [reply]
@jonkoops Yes came across this too, thanks. 
Also for some of the tests intermittently getting below error:
```
-- [ERROR] org.keycloak.testsuite.x509.X509OCSPResponderTest.loginFailedOnOCSPResponderRevocationCheckWithoutCA
-- [ERROR]   Run 1: X509OCSPResponderTest>AbstractKeycloakTest.beforeAbstractKeycloakTest:184->AbstractX509AuthenticationTest.importTestRealms:316->AbstractKeycloakTest.importTestRealms:392->AbstractKeycloakTest.importRealm:512 » Processing jakarta.ws.rs.NotAuthorizedException: HTTP 401 Unauthorized
```
Any help on this please?
